app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -718,8 +718,7 @@ def rbac_edit_save_or_add(what, rbac_suffix = what)
              when :user  then @edit[:user_id] ? User.find_by_id(@edit[:user_id]) : User.new
              end
 
-    send("rbac_#{what}_set_record_vars", record)
-    send("rbac_#{what}_validate?")
+    send("rbac_#{what}_set_record_vars", record) if send("rbac_#{what}_validate?")
```

Comment:
@skateman just adding this IF isn't going to stop feature changes from being changed and saved if `.valid?` on the next line is false (i.e. the `record.name` is blank).  Apparently, setting `record.miq_product_features` in the `rbac_role_set_record_vars` method saves the relationships to the DB.  So, we need to either have both the features relationships and the record changes saved or neither.

@h-kataria is looking into if doing this in a transaction will work here, so expect a comment or code snippet soon.